sunpy/map/tests/test_plotting.py

```diff
@@ -92,10 +93,26 @@ def test_rectangle_aia171(aia171_test_map):
     aia171_test_map.plot()
     bottom_left = SkyCoord(
         0 * u.arcsec, 0 * u.arcsec, frame=aia171_test_map.coordinate_frame)
+    top_right = SkyCoord(
+        100 * u.arcsec, 100 * u.arcsec, frame=aia171_test_map.coordinate_frame)
     w = 100 * u.arcsec
     h = 100 * u.arcsec
     aia171_test_map.draw_rectangle(bottom_left, w, h)
 
+def test_rectangle_aia171_compatibility(aia171_test_map):
+    bottom_left = SkyCoord(
+        0 * u.arcsec, 0 * u.arcsec, frame=aia171_test_map.coordinate_frame)
+    top_right = SkyCoord(
+        100 * u.arcsec, 100 * u.arcsec, frame=aia171_test_map.coordinate_frame)
+    w = 100 * u.arcsec
+    h = 100 * u.arcsec
+    with pytest.deprecated_call():
+        aia171_test_map.draw_rectangle(bottom_left, w, h)
+
+    assert str(aia171_test_map.draw_rectangle(bottom_left, w, h)[0]) == str(aia171_test_map.draw_rectangle(bottom_left, top_right=top_right)[0])
+
+    with pytest.raises(TypeError):
+        aia171_test_map.draw_rectangle(bottom_left, top_right)
```

Comment:
Because of the ordering of the parameters. If you don't pass `top_right` as a named parameter,
it binds to `width` which can only take `units.deg` values and not `SkyCoords`.

Reply:
oh of course! :man_facepalming: